### ISSUE TITLE
Fix trait std::str::FromStr impl

### ISSUE THREAD

**Issue by toqueteos** [PistonDevelopers/gfx_voxel#34]
The [FromStr](http://doc.rust-lang.org/std/str/trait.FromStr.html) trait changed, it now returns a `Result` instead of `Option` so it does need a inner `Err` type.

Any suggestions for `impl FromStr for Face` in cube.rs?

@bvssvni @eddyb

**Comment by toqueteos** [comment]
Related to https://github.com/PistonDevelopers/hematite/issues/182

**Comment by toqueteos** [comment]
I've been [digging on the std](http://doc.rust-lang.org/src/std/old_io/net/ip.rs.html#362-373) and using an empty struct as error seems like a good way to go. We were just returning None if the face wasn't valid.

**Comment by toqueteos** [comment]
Proposal:

```rust
#[derive(Debug, Clone, PartialEq, Copy)]
pub struct ParseError;

impl FromStr for Face {
    type Err = ParseError;
    fn from_str(s: &str) -> Result<Face> {
        match s {
            "down" => Ok(Down),
            "up" => Ok(Up),
            "north" => Ok(North),
            "south" => Ok(South),
            "west" => Ok(West),
            "east" => Ok(East),
            _ => return Err(ParseError)
        }
    }
}
```

**Comment by kvark** [comment]
Similar fixes went in for other libraries, including `gl_generator` and gfx-rs. Please go ahead and submit a PR (your proposal looks good!).

**Comment by kvark** [comment]
Fixed by #35
